### ISSUE TITLE
depz --outdated crashes if repo is in detached head state

### ISSUE THREAD

**Issue by AbePralle** [MurphyMc/depz#5]
STR:
- `depz --init` an existing repo.
- `cd repo && git checkout HEAD^ && cd ..`
- `depz --outdated` will crash.

LOG:
```
DEBUG:depz:Adding . (1 file(s))
DEBUG:depz:Running function early_sanity_check
DEBUG:depz:Running function sanity_check
DEBUG:depz:Running function show_outdated
ERROR:depz:While processing Plasmacore:
Traceback (most recent call last):
  File "/Users/abe/Library/Python/3.7/lib/python/site-packages/depz/depz.py", line 872, in for_each_func
    f(rname,rr, **kw)
  File "/Users/abe/Library/Python/3.7/lib/python/site-packages/depz/depz.py", line 1245, in do_show_outdated
    compares = [r+"/"+current_branch for r in remotes]
  File "/Users/abe/Library/Python/3.7/lib/python/site-packages/depz/depz.py", line 1245, in <listcomp>
    compares = [r+"/"+current_branch for r in remotes]
TypeError: can only concatenate str (not "NoneType") to str
```

**Comment by MurphyMc** [comment]
I wonder what the right behavior is in this case.

Maybe just a warning/error that it can't check for outdatedness when you're not on a branch?

**Comment by AbePralle** [comment]
Sounds good to me.

**Comment by MurphyMc** [comment]
@AbePralle is this resolved now?

**Comment by AbePralle** [comment]
Resolved in [bfd10d9](https://github.com/MurphyMc/depz/commit/bfd10d9d8a2d59e459520d3595ffad0a4e630ca3).
